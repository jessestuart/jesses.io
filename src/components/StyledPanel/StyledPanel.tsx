import classNames from 'classnames'
import React from 'react'
import styled from 'styled-components'

import colors from 'utils/colors'

const StyledPanelContainer = styled.section`
  background-color: ${colors.secondary.light0};
  box-shadow: rgba(25, 17, 34, 0.1) 0px 3px 10px;
  @media (max-width: 45em) {
    margin: 1rem !important;
    padding: 1rem 2rem !important;
  }
`

interface Props {
  children: any
  className?: string
  maxWidth?: number
}

const StyledPanel = ({ children, className, maxWidth = 8 }: Props) => (
  <StyledPanelContainer
    className={classNames(
      `br2 center mv5-ns mw7 mw${maxWidth}-ns pa5-ns`,
      className,
    )}
    style={{ transition: 'all 0.5s' }}
  >
    {children}
  </StyledPanelContainer>
)

export default StyledPanel
