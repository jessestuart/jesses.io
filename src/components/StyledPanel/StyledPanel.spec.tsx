import React from 'react'
import renderer from 'react-test-renderer'
import StyledPanel from './StyledPanel'

test('Render StyledPanel component.', () => {
  const tree = renderer
    .create(<StyledPanel>Basic Styled Panel</StyledPanel>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render StyledPanel with custom styling.', () => {
  const tree = renderer
    .create(<StyledPanel className="w-100">Custom Styled Panel</StyledPanel>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
