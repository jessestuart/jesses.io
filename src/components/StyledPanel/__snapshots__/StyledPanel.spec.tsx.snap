// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render StyledPanel component. 1`] = `
<section
  className="br2 center mv5-ns mw7 mw8-ns pa5-ns sc-bdVaJa diWlBr"
  style={
    Object {
      "transition": "all 0.5s",
    }
  }
>
  Basic Styled Panel
</section>
`;

exports[`Render StyledPanel with custom styling. 1`] = `
<section
  className="br2 center mv5-ns mw7 mw8-ns pa5-ns w-100 sc-bdVaJa diWlBr"
  style={
    Object {
      "transition": "all 0.5s",
    }
  }
>
  Custom Styled Panel
</section>
`;
